scripts-dev/make_full_schema.sh

```diff
@@ -162,12 +162,23 @@ else
 fi
 
 # Delete schema_version, applied_schema_deltas and applied_module_schemas tables
+# Also delete any shadow tables from fts4
 # This needs to be done after synapse_port_db is run
 echo "Dropping unwanted db tables..."
 SQL="
 DROP TABLE schema_version;
 DROP TABLE applied_schema_deltas;
 DROP TABLE applied_module_schemas;
+DROP TABLE event_search_content;
+DROP TABLE event_search_segments;
+DROP TABLE event_search_segdir;
+DROP TABLE event_search_docsize;
+DROP TABLE event_search_stat;
+DROP TABLE user_directory_search_content;
+DROP TABLE user_directory_search_segments;
+DROP TABLE user_directory_search_segdir;
+DROP TABLE user_directory_search_docsize;
+DROP TABLE user_directory_search_stat;
```

Comment:
I haven't tested these changes as the script seems to fail for me before it gets there. Also is there a way to run without a postgres username? (I had to comment that part of the shell script out)


```
$ ./scripts-dev/make_full_schema.sh -o ./
Creating port tables...
Setting up sequence generators...
scripts/synapse_port_db:630: RuntimeWarning: coroutine 'DatabasePool.runInteraction' was never awaited
  await self._setup_user_id_seq()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
2021-01-04 17:48:57,303 - synapse_port_db - 704 - ERROR -
Traceback (most recent call last):
  File "scripts/synapse_port_db", line 631, in run
    await self._setup_events_stream_seqs()
  File "scripts/synapse_port_db", line 910, in _setup_events_stream_seqs
    return await self.postgres_store.db_pool.runInteraction(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 632, in runInteraction
    result = await self.runWithConnection(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 712, in runWithConnection
    return await make_deferred_yieldable(
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 707, in inner_func
    return func(db_conn, *args, **kwargs)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 510, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "scripts/synapse_port_db", line 907, in _setup_events_stream_seqs_set_pos
    (curr_backward_id + 1,),
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
Traceback (most recent call last):
  File "scripts/synapse_port_db", line 631, in run
    await self._setup_events_stream_seqs()
  File "scripts/synapse_port_db", line 910, in _setup_events_stream_seqs
    return await self.postgres_store.db_pool.runInteraction(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 632, in runInteraction
    result = await self.runWithConnection(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 712, in runWithConnection
    return await make_deferred_yieldable(
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 707, in inner_func
    return func(db_conn, *args, **kwargs)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 510, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "scripts/synapse_port_db", line 907, in _setup_events_stream_seqs_set_pos
    (curr_backward_id + 1,),
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```